src/libPMacc/include/cuSTL/container/allocator/compile-time/SharedMemAllocator.hpp

```diff
@@ -42,11 +42,23 @@ struct SharedMemAllocator<Type, Size, 1, uid>
     BOOST_STATIC_CONSTEXPR int dim = 1;
     typedef cursor::CT::BufferCursor<type, math::CT::UInt32<> > Cursor;
 
-    __device__ static Cursor allocate()
+protected:
```

Comment:
changes in this file are independent of the new allocator `LazySharedMemAllocator` please open a new refactoring pull request